engine/src/test/scala/cromwell/engine/workflow/WorkflowDockerLookupActorSpec.scala

```diff
@@ -0,0 +1,391 @@
+        case msg: WorkflowDockerLookupFailure if msg.reason.getMessage == expectedMessage => msg.request
+      }
+    }
+
+    Set(latestRequest, olderRequest) should equal(failedRequests)
+
+    // Try again.  The hashing actor should receive the latest message and this time won't time out.
+    lookupActor ! latestRequest
+    lookupActor ! olderRequest
+    dockerHashingActor.expectMsg(latestRequest)
+    val latestResponse = DockerHashSuccessResponse(DockerHashResult("md5", "AAAAAAAA"), latestRequest)
+    dockerHashingActor.reply(latestResponse)
+
+    val hashResponses = Set(1, 2) map { _ =>
+      expectMsgPF(2 seconds) {
+        case msg: DockerHashSuccessResponse => msg
+      }
+    }
+
+    // Success after transient timeout failures:
+    hashResponses should equal(Set(latestResponse, olderResponse))
+  }
+
+  it should "respond appropriately to docker hash lookup failures" in {
+    val latestDockerId = DockerImageIdentifier.fromString("ubuntu:latest").get.asInstanceOf[DockerImageIdentifierWithoutHash]
+    val olderDockerId = DockerImageIdentifier.fromString("ubuntu:older").get.asInstanceOf[DockerImageIdentifierWithoutHash]
+
+    val workflowId = WorkflowId.randomId()
+    val dockerHashingActor = TestProbe()
+
+    val lookupActor = TestActorRef(WorkflowDockerLookupActor.props(workflowId, dockerHashingActor.ref, StartNewWorkflow))
+    val latestRequest = DockerHashRequest(latestDockerId)
+    val olderRequest = DockerHashRequest(olderDockerId)
+
+    lookupActor ! latestRequest
+    lookupActor ! olderRequest
+
+    // The WorkflowDockerLookupActor should not have the hash for this tag yet and will need to query the dockerHashingActor.
+    dockerHashingActor.expectMsg(latestRequest)
+    dockerHashingActor.expectMsg(olderRequest)
+    val latestSuccessResponse = DockerHashSuccessResponse(DockerHashResult("md5", "AAAAAAAA"), latestRequest)
+    val olderFailedResponse = DockerHashFailedResponse(new RuntimeException("Lookup failed"), olderRequest)
+
+    dockerHashingActor.reply(latestSuccessResponse)
+    dockerHashingActor.reply(olderFailedResponse)
+
+    val mixedResponses = Set(1, 2) map { _ =>
+      expectMsgPF(2 seconds) {
+        case msg: DockerHashSuccessResponse => msg
+        // Scoop out the request here since we can't match the exception on the whole message.
+        case msg: WorkflowDockerLookupFailure if msg.reason.getMessage == "Failed to get docker hash for library/ubuntu:older Lookup failed" => msg.request
+      }
+    }
+
+    Set(latestSuccessResponse, olderRequest) should equal(mixedResponses)
+
+    // Try again, I have a good feeling about this.
+    lookupActor ! olderRequest
+    dockerHashingActor.expectMsg(olderRequest)
+    val olderSuccessResponse = DockerHashSuccessResponse(DockerHashResult("md5", "BBBBBBBB"), olderRequest)
+    dockerHashingActor.reply(olderSuccessResponse)
+    expectMsg(olderSuccessResponse)
+  }
+
+  it should "try to look up hashes on restart" in {
```

Comment:
I'd make this statement stronger, `it should "reuse previously looked up hashes following a restart"`